src/Microsoft.ApplicationInsights.AspNet/TelemetryInitializers/ComponentVersionTelemetryInitializer.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace Microsoft.ApplicationInsights.AspNet.TelemetryInitializers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Channel;
+    using DataContracts;
+    using Microsoft.AspNet.Http;
+    using Microsoft.Extensions.Configuration;
+
+    /// <summary>
+    /// A telemetry initializer that populates telemetry.Context.Component.Version to the value read from project.json
+    /// </summary>
+    public class ComponentVersionTelemetryInitializer : TelemetryInitializerBase
+    {
+        private const string _versionConfigurationOption = "dependencies:Microsoft.ApplicationInsights.AspNet";
```

Comment:
I think we need this: https://github.com/Microsoft/ApplicationInsights-aspnetcore/blob/master/test/WebApiShimFw46.FunctionalTests/project.json#L3